course/Program.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace Course
+{
+    class Course
+    {
+        static void Main()
+        {
+            var stack = new Stack<string>();
+            stack.Push("A");
+            stack.Push("B");
+            stack.Push("C");
+            var arr = new string[] {};
+            stack.CopyTo(ref arr);
+            PrintArray(arr);
+        }
+
+        static void PrintArray<T>(T[] arr)
+        {
+            foreach (T item in arr)
+            {
+                Console.WriteLine(item);
+            }
+        }
+    }
+
+    public class Stack<T>
+    {
+        public class StackItem
+        {
+            public T Value { get; set; }
+            public StackItem Previous { get; set; }
+        }
+
+        private StackItem _last;
+        private int _count;
+
+        public void Push(T obj)
+        {
+            var stackItem = new StackItem
+            {
+                Value = obj
+            };
+
+            if (_last == null)
+            {
+                _last = stackItem;
+            }
+            else
+            {
+                stackItem.Previous = _last;
```

Comment:
if you leave code in 'else' branch without if/else - all will work as previously